frontend/model/contracts/group.js

```diff
@@ -697,7 +767,25 @@ DefineContract({
         // }
       }
     },
-
+    'gi.contracts/group/resetMonth': {
+      validate: optional(string),
+      process ({ meta }, { state, getters }) {
+        let lastEvent = state.distributionEvents[state.distributionEvents.length - 1]
+        const currentCycle = cycleAtDate(meta.createdDate, state.distributionCycleStartDate)
+        // Add 'startCycleEvent' events for every month missed.
+        while (Math.floor(lastEvent.data.cycle) < Math.floor(currentCycle)) {
+          const monthlyCycleEvent = {
+            type: 'startCycleEvent',
+            data: {
+              cycle: Math.floor(lastEvent.data.cycle + 1),
+              latePayments: [] // List to be populated later, by the events-parser
+            }
+          }
+          state.distributionEvents.push(monthlyCycleEvent)
+          lastEvent = monthlyCycleEvent
+        }
```

Comment:
Convert this whole thing to an `if` statement per call.

Reply:
Changed.